### PR TITLE
Fuse confirm_and_abort_neg into confirm_and_abort

### PR DESCRIPTION
**Pull request by pitag-ha**
I'm just restructuring the confirm_and_abort for negative default behavior according to @gpetiot 's suggestion in #231.

### CODE REVIEW

File: lib/prompt.ml [hunk 1]
```diff
@@ -21,14 +21,9 @@ let rec loop ~question ~default_answer =
              for the default");
       loop ~question ~default_answer
 
-let confirm ~question ~yes =
-  if yes then true else loop ~question ~default_answer:Yes
+let confirm ~question ~yes ~default_answer =
+  if yes then true else loop ~question ~default_answer
```
Comment (gpetiot) [inline lib/prompt.ml:25]: I would make `default_answer` optional and make the default behavior to `Yes` since it's the main usecase (and adding the argument everywhere generates an unnecessary diff)

Comment (pitag-ha) [reply lib/prompt.ml:25]: Do I have to add the unit as a last parameter for that to work? If so, I think the diff would look similar :/ Or is there some other trick to make that work that I'm not aware of?

Comment (gpetiot) [reply lib/prompt.ml:25]: Oh yes I didn't pay attention to that

Comment (pitag-ha) [reply lib/prompt.ml:25]: Would you still do that? Making the `default_answer` optional (defaulting to `Yes`) and adding the unit as positional parameter I mean?

Comment (gpetiot) [reply lib/prompt.ml:25]: No I guess I would do like you did

File: lib/prompt.mli [hunk 2]
```diff
@@ -1,17 +1,19 @@
-val confirm : question:('a, unit) Logs.msgf -> yes:bool -> bool
-(** Promtps the user for confirmation. [confirm ~question ~yes] uses the message
-    formatting function [question] to format and log a message with the app
-    level and wait for a yes or no answer from the user. Returns [true] for yes.
-    Defaults to yes if the user just press enter. If [yes] then just skip the
-    prompt and returns [true]. E.g.
+type default = Yes | No
```
Comment (NathanReb) [inline lib/prompt.mli:1]: Not that it matters much but how about naming this type `answer` instead? I think it makes more sense.
